File: src/server/ai/providers/cerebras.ts
import { AIAssistantConfig } from "../config";
import { OpenAICompatibleProvider } from "./base";

export const cerebrasProvider = new OpenAICompatibleProvider(
  "cerebras",
  {
    apiKey: AIAssistantConfig.cerebrasApiKey,
    baseUrl: "https://api.cerebras.ai/v1/chat/completions",
    model: AIAssistantConfig.cerebrasModel,
    timeoutMs: AIAssistantConfig.requestTimeoutMs,
    maxRetries: AIAssistantConfig.maxRetries,
  }
);

File: ai-assistant/providers/cerebras.py
import os
from .base import OpenAICompatibleProvider

cerebras_provider = OpenAICompatibleProvider(
    name="cerebras",
    api_key=os.getenv("CEREBBRAS_API_KEY", ""),
    base_url="https://api.cerebras.ai/v1/chat/completions",
    model=os.getenv("CEREBBRAS_MODEL", "gpt-oss-120b"),
)
